work/worker.go

```diff
@@ -1043,10 +1112,40 @@ func (env *Task) commitBundleTransaction(bundle *builder.Bundle, bc BlockChain,
 	env.size += totalTxSize
 	env.txs = append(env.txs, txs...)
 	env.receipts = append(env.receipts, receipts...)
-
+	// blob related env are updated
+	env.sidecars = append(env.sidecars, sidecars...)
+	env.blobs += blobs
 	return nil, nil, logs
 }
 
+// commitBlobTransaction is a wrapper for executeTx.
+// It updates the blob-specific env for the argument tx and returns tx.WithoutBlobTxSidecar().
+func (env *Task) commitBlobTransaction(tx *types.Transaction, sidecars *[]*types.BlobTxSidecar, blobs *int, executeTx func() (error, *types.Receipt)) (err error, executedTx *types.Transaction, receipt *types.Receipt) {
+	sc := tx.BlobTxSidecar()
+	if sc == nil {
+		panic("blob transaction without blobs in miner")
+	}
+	// Checking against blob gas limit: It's kind of ugly to perform this check here, but there
+	// isn't really a better place right now. The blob gas limit is checked at block validation time
+	// and not during execution. This means core.ApplyTransaction will not return an error if the
+	// tx has too many blobs. So we have to explicitly check it here.
+	maxBlobs := eip4844.MaxBlobsPerBlock(env.config, env.header.Number)
+	if env.blobs+len(sc.Blobs) > maxBlobs {
+		return errors.New("max data blobs reached"), tx, nil
+	}
+
+	err, receipt = executeTx()
+	if err != nil {
+		return err, tx, nil
+	}
+
+	txNoBlob := tx.WithoutBlobTxSidecar()
```

Comment:
Why do you purge blob?

Reply:
You're right, it shouldn't have been deleted here.

Sidecars are eliminated by dividing responsibilities between `worker` and the `blockchain`.

- In `blockchain.WriteBlockWithState` to prevent blobs from being included in block writes.
- In `worker.wait()` to prevent a consensus block containing a sidecar from being broadcast. KIP-279 only defines a consensus proposal.

This allows us to keep sidecar in the consensus proposal while preserving WithoutSidecar from the db and block broadcast. 
Fixed at [f923085](https://github.com/kaiachain/kaia/pull/683/commits/f923085560b717ad7d2c6f0404c198db702c0135), Thanks!